src/webgpu/shader/execution/expression/call/builtin/texture_utils.spec.ts

```diff
@@ -151,3 +154,125 @@ g.test('readTextureToTexelViews')
       assert(errors.length === 0, errors.join('\n'));
     }
   });
+
+function validateWeights(stage: string, builtin: string, weights: number[]) {
+  const kNumMixSteps = weights.length - 1;
+  const showWeights = () => `
+${weights.map((v, i) => `${i.toString().padStart(2)}: ${v}`).join('\n')}
+
+e = expected
+A = actual
+${graphWeights(32, weights)}
+`;
+
+  assert(
+    weights[0] === 0,
+    `stage: ${stage}, ${builtin}, weight 0 expected 0 but was ${weights[0]}\n${showWeights()}`
+  );
+  assert(
+    weights[kNumMixSteps] === 1,
+    `stage: ${stage}, ${builtin}, top weight expected 1 but was ${
+      weights[kNumMixSteps]
+    }\n${showWeights()}`
+  );
+
+  const dx = 1 / kNumMixSteps;
+  for (let i = 0; i < kNumMixSteps; ++i) {
+    const dy = weights[i + 1] - weights[i];
+    // dy / dx because dy might be 0
+    const slope = dy / dx;
+    assert(
+      slope >= 0,
+      `stage: ${stage}, ${builtin}, weight[${i}] was not <= weight[${i + 1}]\n${showWeights()}`
+    );
+    assert(
+      slope <= 2,
+      `stage: ${stage}, ${builtin}, slope from weight[${i}] to weight[${
+        i + 1
+      }] is > 2.\n${showWeights()}`
+    );
+  }
+
+  assert(
```

Comment:
optional: it might be useful to turn some or all of these `assert`s into `t.expect`s so the test doesn't bail out as soon as it fails. Depends on how spammy that would make the log.

Reply:
done